rustiflow/src/flows/cic_flow.rs

```diff
@@ -533,13 +528,13 @@ impl CicFlow {
     ///
     /// * `timestamp` - The timestamp of the packet or event triggering the update.
     /// * `threshold` - The threshold in microseconds to determine state transitions between active and idle.
-    fn update_active_idle_time(&mut self, timestamp: &Instant, threshold: f64) {
-        if timestamp.duration_since(self.end_active).as_micros() as f64 > threshold {
-            let duration = self.end_active.duration_since(self.start_active);
-            if duration.as_secs_f64() > 0.0 {
-                self.update_active_flow(duration.as_micros() as f64);
+    fn update_active_idle_time(&mut self, timestamp: &DateTime<Utc>, threshold: f64) {
+        if timestamp.signed_duration_since(self.end_active).num_microseconds().unwrap() as f64 > threshold {
+            let duration = self.end_active.signed_duration_since(self.start_active);
+            if !duration.is_zero() {
```

Comment:
Don't know if this would ever occur, but what if the duration is smaller then zero?

Reply:
signed_duration_since: Subtracts another DateTime from the current date and time. This does not overflow or underflow at all.

I expect no problems